Documentation/releases.md

```diff
@@ -0,0 +1,23 @@
+## Making a dex Release
+
+1. Make a new clone of the dex repo:
+
+  ```console
+  $ git clone git@github.com:coreos/dex.git
+  $ cd dex
+  ```
+
+1. Tag with the release name:
+
+   ```console
+   git tag -s v0.4.0
```

Comment:
We should probably make a note that the releaser needs upload their public GPG key to github tho.